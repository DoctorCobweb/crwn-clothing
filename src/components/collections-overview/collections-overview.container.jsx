import { connect } from 'react-redux';
// import { createStucturedSelector } from 'reselect';
import { createStructuredSelector } from 'reselect';
import { compose } from 'redux';

import { selectIsCollectionFetching } from '../../redux/shop/shop.selectors';

import WithSpinner from '../with-spinner/with-spinner.component';
import CollectionsOverview from './collections-overview.component';

const mapStateToProps = createStructuredSelector({
  isLoading: selectIsCollectionFetching,
});


// send CollectionsOverview into WithSpinner, then that result into connect(mapStateToProps)
const CollectionsOverviewContainer = compose(
  connect(mapStateToProps),
  WithSpinner
)(CollectionsOverview);

export default CollectionsOverviewContainer;
