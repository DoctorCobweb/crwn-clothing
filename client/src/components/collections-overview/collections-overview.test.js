import React from 'react';
import { shallow } from 'enzyme';

import { CollectionsOverview } from './collections-overview.component';

describe('CollectionsOverview component', () => {
  let wrapper;
  let mockProps;

  beforeEach(()=> {
    mockProps = {
      collections: [
        {
          id: 0,
        },
        {
          id: 1,
        },
        {
          id: 2,
        }
      ]
    };
    wrapper = shallow(<CollectionsOverview {...mockProps} />)
  });

  it('should render CollectionsOverviewContainer componnet', () => {
    expect(wrapper.debug()).toMatchSnapshot();
  });

  it('should render all the CollectionPreview components', () => {
    expect(wrapper.find('CollectionsOverviewContainer').children().length).toBe(mockProps.collections.length)
  });

});