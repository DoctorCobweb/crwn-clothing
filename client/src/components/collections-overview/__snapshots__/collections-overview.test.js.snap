// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollectionsOverview component should render CollectionsOverviewContainer componnet 1`] = `
"<CollectionsOverviewContainer>
  <withRouter(CollectionPreview) />
  <withRouter(CollectionPreview) />
  <withRouter(CollectionPreview) />
</CollectionsOverviewContainer>"
`;
